src/v1/sources/shopify/webhookTransformations/serverSideTransform.js

```diff
@@ -154,9 +155,9 @@ const processEvent = async (inputEvent, metricMetadata) => {
   }
   message = removeUndefinedAndNullValues(message);
 
-  // if the message payload contains both anonymousId and userId, hence the user is identified
+  // if the message payload contains both anonymousId and userId or contains customer object, hence the user is identified
   // then create an identify event by multiplexing the original event and return both the message and identify event
-  if (message.anonymousId && message.userId) {
+  if ((message.anonymousId && message.userId) || customer) {
```

Comment:
What happens when we have `message.anonymousId && message.userId` but no `customer`?

The Identify call does not get any traits details

Reply:
I am not sure if that could happen, we mainly wanted the reverse, i.e what if userId is not there in redis, but shopify will be sending customer object in the ecom event, it would be used to send a call with anonymousId and customer details in traits.

Yes for mentioned case, traits will be empty

Reply:
Why not keep condition only with `customer` object presence @krishna2020 WDYT?

Reply:
There are 2 use-cases here:
- stitch anonmyousId profile (as this gets created first) to userId (when customer identified)
- update user profile attributes whenever traits are available in the payload and this definitely needs one of anonId/userId

Even if identify call doesn't have traits, that shouldn't be an issue.
And with the current condition, if both userId/anonId are not available in payload, there's no use of customer object --> and creatIdentifyEvent doesn't has any null check on userId & anonId

Reply:
if ((userId && anonId) || (userId && customer) || (anonId && customer)) should be right condition
Additionally, creatIdentifyEvent shouldn't set userId/anonId if there are missing

Reply:
Updated the condition to above mentioned and also added check within `createIdentifyEvent` for anonymousIs and userId